deflate_quick.c

```diff
@@ -85,9 +85,8 @@ Z_INTERNAL block_state deflate_quick(deflate_state *s, int flush) {
 
         if (LIKELY(s->lookahead >= MIN_MATCH)) {
             hash_head = functable.quick_insert_string(s, s->strstart);
-            dist = s->strstart - hash_head;
 
-            if (dist > 0 && dist < MAX_DIST(s)) {
+            if (hash_head != 0 && (dist = s->strstart - hash_head) < MAX_DIST(s)) {
```

Comment:
I still think assignment inside of conditional is a bad thing, even if the alternative makes the patch longer... We should always prefer readability over patch length...

Reply:
I know you have said the same thing before about this particular code, but I keep going back to it. I like the assignment here. The code more closely matches the other if statements in the other algorithms.
```
if (hash_head != 0 && s->strstart - hash_head <= MAX_DIST(s)) {
```
And..
```
if (hash_head != 0 && (dist = s->strstart - hash_head) < MAX_DIST(s)) {
```

Reply:
I know that... I also know that sometimes there is/can be differences that are justified...